sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CallToAsyncMethodShouldNotBeBlocking.cs

```diff
@@ -71,17 +71,13 @@ public sealed class CallToAsyncMethodShouldNotBeBlocking : SonarDiagnosticAnalyz
                 ["Sleep"] = new[] { "Thread.Sleep", "await Task.Delay" },
             };
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                ReportOnViolation,
-                SyntaxKind.SimpleMemberAccessExpression);
-        }
+        protected override void Initialize(SonarAnalysisContext context) =>
```

Comment:
Regarding `InvalidMemberAccess` vs `MemberNameToMessageArguments`, I don't like that there are two duplicate sets of initialization that hopes to contain same keys. 
Can you please either:
- sort the members to be in the same alphabetical order and put the declarations together (moving `TaskThreadPoolCalls` before or after)
or
- create some local Config class that would contain all information needed for single method name. And initialize it something like 
`["GetResult"] = new MethodConfig("Task.GetAwaiter.GetResult", "await", KnownType.System_Runtime_CompilerServices_TaskAwaiter, KnownType.System_Runtime_CompilerServices_TaskAwaiter_TResult)`

Reply:
Moved MemberNameToMessageArguments up and did the ordering but I would avoid spreading this commit even more. I suggest to start with baby steps.